src/function_space/shape_function_generator.jl

```diff
@@ -0,0 +1,411 @@
+# This file gathers shape function generation utils
+
+using Symbolics
+using LinearAlgebra
+using ForwardDiff
+
+@variables x[1:3]
+
+abstract type GeneratedFiniteElement <: AbstractFunctionSpaceType end
+struct LagrangeGenerated <: GeneratedFiniteElement end
+struct HermiteGenerated <: GeneratedFiniteElement end
+
+function FunctionSpace(::Val{:LagrangeGenerated}, degree::Integer)
+    FunctionSpace(LagrangeGenerated(), degree)
+end
+function FunctionSpace(::Val{:HermiteGenerated}, degree::Integer)
+    FunctionSpace(HermiteGenerated(), degree)
+end
+
+basis_functions_style(::FunctionSpace{<:LagrangeGenerated}) = NodalBasisFunctionsStyle()
+
+abstract type LinearForm end
+
+"""
+  Defines a point evaluation linear functional l(f) = f(x) (x is fixed)
+"""
+struct PointEvaluationForm <: LinearForm
+    x::Vector{Float64}
+end
+
+function (l::PointEvaluationForm)(f::Function)
+    return f(l.x)
+end
+
+"""
+  Defines a derivative point evaluation linear functional l(f) = f'(x) (x is fixed)
+"""
+struct PointDerivativeForm <: LinearForm
+    x::Vector{Float64}
+end
+
+function (l::PointDerivativeForm)(f::Function)
+    dfdx = x -> ForwardDiff.gradient(f, x)[1]
+    return dfdx(l.x)
+end
+
+"""
+  Returns the linear forms for continous Lagrange shape functions on a line => evaluation of f at nodes
+"""
+function Forms(::LagrangeGenerated, ::Line, deg)
```

Comment:
function starting with a capital letter are reserved for struct constructors. Maybe you could rename this as something like `build_forms_array`?